pattern-library/sass/patterns/_dropdown-menu.scss

```diff
@@ -0,0 +1,173 @@
+// ------------------------------
+// edX Pattern Library: Utilities - Dropdown Menu
+//
+// About: Contains base styling for alerts.
+// ----------------------------
+
+// #CONFIG
+// #UTILITIES
+// #GENERAL
+// #INDIVIDUAL CASES
+
+
+// ----------------------------
+// #CONFIG
+// ----------------------------
+$dropdown-menu-button-font-size: font-size(small) !default;
+$dropdown-menu-list-font-size: font-size(base) !default;
+$dropdown-menu-z-index: 1001 !default;
+$dropdown-menu-button-text-align: left !default;
+$dropdown-menu-list-text-align: left !default;
+$dropdown-menu-width: 170px !default;
+
+$dropdown-menu-focus-border-color: palette(primary, light) !default;
+$dropdown-link-hover-color: palette(primary, light) !default;
+$dropdown-menu-list-hover-color: palette(grayscale, x-dark) !default;
+$dropdown-menu-font-color: $black !default;
+$dropdown-menu-list-font-color: palette(primary, base) !default;
+$dropdown-menu-background-color: $white !default;
+$dropdown-menu-border-color: $gray !default;
+$dropdown-menu-list-border-color: $gray-cool !default;
+$dropdown-menu-list-border-bottom: 1px dotted $dropdown-menu-list-border-color !default;
+
+// ----------------------------
+// #UTILITIES
+// ----------------------------
+
+
+// ----------------------------
+// #GENERAL
+// ----------------------------
+
+.dropdown-menu-container {
+  height: 40px;
+  position: relative;
+  display: inline;
+}
+
+.dropdown-menu {
+    list-style-type: none;
+}
+
+.menu-title {
+  font: {
+    family: $font-family-sans-serif;
+    size: $dropdown-menu-button-font-size;
+    weight: font-weight(bold);
+  }
+  color: $dropdown-menu-font-color;
+  margin: {
+    left: 15px;
+    right: 2px;
+  }
+  line-height: 40px;
+
+  &:active,
+  &:hover,
+  &:focus {
+    color: $dropdown-link-hover-color;
+  }
+}
+
+.menu-image {
+  @include margin-right(6px);
+  width: 40px;
+  height: 40px;
+}
+
+.button {
+  @include text-align($dropdown-menu-button-text-align);
+  background: none;
+  width: 30px;
+  height: 30px;
+  padding: 0;
+  margin-top: 6px;
+  border: none;
+  vertical-align: top;
+  position: relative;
+
+  &:after {
+    content: '';
+    width: 0;
+    height: 0;
+    border: 5px solid transparent;
+    border-top-color: $dropdown-menu-font-color;
+    border-top-width: 6px;
+    position: absolute;
+    top: 11px;
+    right: 10px;
+  }
+
+  &:active,
+  &:hover,
+  &:focus {
+    &:after {
+      border-top-color: $dropdown-menu-focus-border-color;
+    }
+  }
+}
+
+.dropdown-menu {
+  position: absolute;
+  top: 35px;
+  right: 0;
+  background: $dropdown-menu-background-color;
+  z-index: $dropdown-menu-z-index;
+  border-radius: 4px;
+  box-shadow: 0 2px 24px 0 rgba(0,0,0,0.3);
+  border: 1px solid $dropdown-menu-border-color;
+  margin-top: 0;
+  padding: 5px 10px;
+  width: $dropdown-menu-width;
+
+  &:before {
```

Comment:
I didn't realize this was initially for the caret.  A comment would be enlightening!